src/uu/ps/src/picker.rs

```diff
@@ -70,3 +75,47 @@ fn cmd(proc_info: RefCell<ProcessInformation>) -> String {
 fn ucmd(proc_info: RefCell<ProcessInformation>) -> String {
     proc_info.borrow_mut().status().get("Name").unwrap().into()
 }
+
+#[test]
+fn test_time() {
+    {
+        let cumulative_cpu_time = {
+            let utime = 29i64;
+            let stime = 18439i64;
+            (utime + stime) / 100
+        };
+
+        let days = cumulative_cpu_time / (3600 * 24);
+        let hours = (cumulative_cpu_time % (3600 * 24)) / 3600;
+        let minutes = (cumulative_cpu_time % 3600) / 60;
+        let seconds = cumulative_cpu_time % 60;
+
+        let result = if days != 0 {
+            format!("{:02}-{:02}:{:02}:{:02}", days, hours, minutes, seconds)
+        } else {
+            format!("{:02}:{:02}:{:02}", hours, minutes, seconds)
+        };
+
+        assert!(result == "00:03:04");
+    }
+    {
+        let cumulative_cpu_time = {
+            let utime = 1145141919i64;
+            let stime = 810i64;
+            (utime + stime) / 100
+        };
+
+        let days = cumulative_cpu_time / (3600 * 24);
+        let hours = (cumulative_cpu_time % (3600 * 24)) / 3600;
+        let minutes = (cumulative_cpu_time % 3600) / 60;
+        let seconds = cumulative_cpu_time % 60;
+
+        let result = if days != 0 {
+            format!("{:02}-{:02}:{:02}:{:02}", days, hours, minutes, seconds)
+        } else {
+            format!("{:02}:{:02}:{:02}", hours, minutes, seconds)
+        };
+
+        assert!(result == "132-12:57:07");
+    }
+}
```

Comment:
As I said, you can coding your logic only once and make sure it's covered by unit tests XD.